Marlin/src/module/planner.h

```diff
@@ -559,9 +559,37 @@ class Planner {
       #if ENABLED(FWRETRACT)
         apply_retract(raw);
       #endif
+
       #if IS_KINEMATIC
+        #if HAS_FEEDRATE_SCALING
+          // DELTA/SCARA need to recalculate the feedrate in
+          // order to complete the move in the given time.
+          const float inv_segment_length = 1.0f / millimeters,
+                      inverse_secs = inv_segment_length * fr_mm_s,
+                      oldA = position_float[A_AXIS],
+                      oldB = position_float[B_AXIS]
+                      #if ENABLED(DELTA_FEEDRATE_SCALING)
+                        , oldC = position_float[C_AXIS]
+                      #endif
+                      ;
+        #endif
+
         inverse_kinematics(raw);
-        return buffer_segment(delta[A_AXIS], delta[B_AXIS], delta[C_AXIS], raw[E_AXIS], fr_mm_s, extruder, millimeters);
+
+        const float feedrate = (
+          #if ENABLED(SCARA_FEEDRATE_SCALING)
+            // For SCARA scale the feed rate from mm/s to degrees/s
+            // i.e., Complete the angular vector in the given time.
+            HYPOT(delta[A_AXIS] - oldA, delta[B_AXIS] - oldB) * inverse_secs
+          #elif ENABLED(DELTA_FEEDRATE_SCALING)
+            // For DELTA scale the feed rate from Effector mm/s to Carriage mm/s
+            // i.e., Complete the linear vector in the given time.
+            SQRT(sq(delta[A_AXIS] - oldA) + sq(delta[B_AXIS] - oldB) + sq(delta[C_AXIS] - oldC)) * inverse_secs
+          #else
+            fr_mm_s
+          #endif
+        );
+        return buffer_segment(delta[A_AXIS], delta[B_AXIS], delta[C_AXIS], raw[E_AXIS], feedrate, extruder, millimeters);
```

Comment:
This is one point where `delta[ABC]` could be copied to `oldA`, `oldB`, `oldC`, but it would be wrong. They should be copied from `a`, `b`, `c` in the `buffer_segment` method instead.